storage/core/src/server/ServerStorage.ts

```diff
@@ -213,6 +210,8 @@ export abstract class ServerStorage
    * @param {BaseDirectory} targetDirectory base directory in the target storage.
    * @param {Function} predicate optional predicate to filter objects to copy. If not specified, all
    * objects from the sourceDirectory will be copied.
+   * @param {Function} renameCallback optional callback to rename objects as they are copied. If
+   * specified, targetDirectory argument will be ignored.
```

Comment:
would it make sense to have either target directory or renameCallback?
`target: BaseDirectory | (objectReference: ObjectReference) => ObjectReference`